src/graph/sampling/randomwalks/frequency_hashmap.cu

```diff
@@ -0,0 +1,439 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file graph/sampling/frequency_hashmap.cu
+ * \brief frequency hashmap - used to select top-k frequency edges of each node
+ */
```

Comment:
We have a `OrderedHashTable` class in `src/runtime/cuda/cuda_hashtable.cuh`.  Could you see if you can utilize that?  My intuition is that you can subclass it and add a new select-top-k interface.